### PR TITLE
Fix(eos_designs): Ignore "overlay_routing_protocol_address_family: ipv6" on l2leaf (#2955)

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Change Summary

Cherry-pick PR #2955

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/DC1-L2LEAF1A.cfg [hunk 1]
```diff
@@ -121,9 +121,11 @@ interface Vlan4091
    no autostate
    ip address 10.255.252.14/31
 !
-ip routing
+ip routing ipv6 interfaces
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/DC1-L2LEAF1A.cfg]: I don't think these changes are correct. Please compare with the `devel` files.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/DC1-L2LEAF1A.cfg [hunk 2]
```diff
@@ -121,9 +121,11 @@ interface Vlan4091
...
 no ip routing vrf MGMT
 !
+ipv6 unicast-routing
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/DC1-L2LEAF1A.cfg]: I don't think these changes are correct. Please compare with the `devel` files.

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py [hunk 3]
```diff
@@ -1941,34 +1941,47 @@ def _overlay_vpn_ipv6(self) -> bool:
         )
 
     @cached_property
-    def overlay(self) -> dict:
+    def overlay_peering_address(self) -> str | None:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py]: This should not be exposed in the generated facts, so it should be prefixed with underscore.

```suggestion
    def _overlay_peering_address(self) -> str | None:
```

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py [hunk 4]
```diff
@@ -1941,34 +1941,47 @@ def _overlay_vpn_ipv6(self) -> bool:
...
 
+        if self.underlay_router is True:
+            overlay_dict.update(
+                {
+                    "peering_address": self.overlay_peering_address,
+                    "evpn_mpls": evpn_mpls,
+                }
+            )
+
+        return overlay_dict
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py]: I don't think you need this change. The entire structure of `overlay` has changed between the versions, so it is not possible to apply the same logic.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/host_vars/DC1-L2LEAF1A.yml [hunk 5]
```diff
@@ -2,3 +2,8 @@
 # Test that having underlay_ipv6 will not require
 # loopback_ipv6_pool for devices that are not underlay_router
 underlay_ipv6: true
+underlay_rfc5549: true
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/host_vars/DC1-L2LEAF1A.yml]: This line is not in the cherry_picked PR, but maybe this actually reveals another issue with l2 leaf and underlay_rfc5549, since you see the ipv6_unicast_routing and ip_routing_ipv6_interfaces being set.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/DC1-L2LEAF1A.cfg (hunk 1 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/DC1-L2LEAF1A.cfg]: Fixed!

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/DC1-L2LEAF1A.cfg (hunk 2 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/DC1-L2LEAF1A.cfg]: Fixed!

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py (hunk 3 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py]: Fixed!

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/host_vars/DC1-L2LEAF1A.yml (hunk 5 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/host_vars/DC1-L2LEAF1A.yml]: Removed the line.

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py (hunk 4 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py]: We have kept this almost same, just updating the "overlay_dict" with latest devel changes, in the end we are not returning "None" (as it is in latest devel), we are returning the same dict what was there, just removing "peering_address" and "evpn_mpls" from it (as it is in latest devel).

Would you like us to change this back to how it was in 3.8?

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py]: Yes I don't think these changes apply for the cherry pick.

Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py]: Ok.. In that case shall I remove [this](https://github.com/aristanetworks/ansible-avd/pull/3043/files#diff-ef9c24ca7d287ca54b0ee4167c4958d47845c503f9ac5db2cb228060144c4915R8) line as well?
otherwise we will get an error similar to :

```
"Unexpected failure during module execution: 'overlay_routing_protocol_address_family: ipv6' is only supported in combination with 'underlay_ipv6: True' and 'underlay_rfc5549: True'"
```

Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py]: Fixed!

**Review by carlbuchmann — DISMISSED** [review]
LGTM

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py [hunk 6]
```diff
@@ -1830,7 +1830,7 @@ def mlag_peer_mgmt_ip(self):
         return None
 
     @cached_property
-    def overlay_routing_protocol_address_family(self):
+    def _overlay_routing_protocol_address_family(self):
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py]: Please revert this change, since it was already there without the underscore, meaning it ends up as `switch.overlay_routing_protocol_address_family` which may be used elsewhere in 3.8.

Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py]: If we remove underscore "_" then we get the following error in eos_designs_unit_tests molecule:

```
  An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible_collections.arista.avd.plugins.plugin_utils.errors.errors.AristaAvdError: 'overlay_routing_protocol_address_family: ipv6' is only supported in combination with 'underlay_ipv6: True' and 'underlay_rfc5549: True'
  ```
  
  In order to fix the error we modified it, is there any other way to do it?

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py]: Yeah because changing this will break that check overall even for other device types.

Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py]: Done :)

**Review by ClausHolbechArista — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM based on CI results!
